src/main/java/nextstep/security/oauth2/OAuth2AuthorizationRequestRedirectFilter.java

```diff
@@ -0,0 +1,46 @@
+package nextstep.security.oauth2;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import nextstep.security.access.PathPatternRequestMatcher;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+public class OAuth2AuthorizationRequestRedirectFilter extends OncePerRequestFilter {
+    private static final String OAUTH_REQUEST_URI_PATTERN = "/oauth2/authorization/*";
+
+    private final OAuth2AuthorizationRequestResolver oAuth2AuthorizationRequestResolver;
+    private final PathPatternRequestMatcher pathPatternRequestMatcher;
```

Comment:
```suggestion
    private final RequestMatcher pathPatternRequestMatcher;
```
잘 추상화되어있으니 구현체에 의존하지 않게 변경할 수 있겠네요!